Add testing for all supported Python and Django versions

jdufresne opened the pull request:
Now include Django 1.11 and Python 3.6 in testing.

Moved to an explicit matrix as there isn't consistent overlap in support
of Python across Django versions.

Added all versions to the trove classifiers in setup.py.

Added pip caching to travis defintion.

nyergler commented:
Thanks for updating the testing matrix. FWIW in other projects I've just been testing with supported Django versions, so I'll probably drop 1.9 out of here at some point soon.

Any specific reason for vendoring in `flatten_to_dict` from Rebar?

jdufresne commented:
I've dropped 1.9 in the latest revision.

I originally added `flatten_to_dict()` as rebar wasn't compatible when using the latest Django. But that appears to be resolved simple by adjusting the test settings. So I've reverted to using the rebar version.

nyergler commented:
Thanks, @jdufresne. Apologies for the delays in merging.
